subcommands/targets/add.go

```diff
@@ -0,0 +1,288 @@
+package targets
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/theupdateframework/notary"
+	tuf "github.com/theupdateframework/notary/tuf/data"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	addTargetType string
+	addTags       string
+	addSrcTag     string
+	addOutFile    string
+	addPrint      bool
+	addDryRun     bool
+)
+
+// Consider moving the `Target` type along with its receivers to the `client` package
+// or do opposite - "hide" it
```

Comment:
OK. I hesitated putting it into the client API because I didn't want to impact the code that is used with the code that may not be used by anyone.
However, the final version seems quite independent from the rest of the code so I think that's fine to move it to the client package.

Reply:
https://github.com/foundriesio/fioctl/pull/198/commits/000b6fd154022972219861897669da77d2504ce3